javaslang/src/main/java/javaslang/concurrent/Future.java

```diff
@@ -309,10 +309,41 @@
      * @throws NullPointerException if one of executorService of computation is null.
      */
     static <T> Future<T> of(ExecutorService executorService, CheckedSupplier<? extends T> computation) {
+        return Future.of(executorService, computation, 0);
+    }
+
+    /**
+     * Starts an asynchronous computation after delaying, backed by the {@link #DEFAULT_EXECUTOR_SERVICE}.
+     * The thread supplied by the {@code DEFAULT_EXECUTOR_SERVICE} will be blocked at least {@code delay} milliseconds.
+     * If the {@code delay} &lt;= 0, the {@code computation} will be executed immediately.
+     *
+     * @param computation A computation.
+     * @param delay       The time in milliseconds to wait before executing the given {@code computation}.
+     * @param <T>         Type of the computation result.
+     * @return A new Future instance.
+     * @throws NullPointerException if computation is null.
+     */
+    static <T> Future<T> of(CheckedSupplier<? extends T> computation, long delay) {
```

Comment:
please express the unit in the name also, people shouldn't have to look in the documentation to see whether `of (..., 1)` sleeps for `1 milllisecond` or `1 second`.
I recommend something like:
```java
static <T> Future<T> of(CheckedSupplier<? extends T> computation, long delayMillis)
```
instead, to mimic e.g.
```java
public static native long currentTimeMillis();
```